picard/pluginmanager.py

```diff
@@ -553,3 +591,24 @@ def _display_update():
             self.query_available_plugins(_display_update)
         else:
             _display_update()
+
+
+class PluginMetaPathFinder(MetaPathFinder):
+    def find_spec(self, fullname, path, target=None):
+        if not fullname.startswith(_PLUGIN_MODULE_PREFIX):
+            return None
+        plugin_name = fullname[len(_PLUGIN_MODULE_PREFIX):]
+        spec = None
```

Comment:
Instead of using `_plugin_dirs` global variable (as it is somehow "private"), we should use the generator wrapping it instead:

```python
paths = tuple(plugin_dirs())
```

and `s/_plugin_dirs/paths/` below.

Reply:
I refactored the class a bit in  https://github.com/metabrainz/picard/pull/2307/commits/70904c11e035c2ad850c2eb3d09509808e25aea9 . I think this makes it easier to understand and extend. It unifies handling of zipped and normal file plugins.